tools/azsdk-cli/specs/1-env-setup-env-verification.spec.md

```diff
@@ -0,0 +1,576 @@
+│  └───────────────┬───────────────────────────┘  │
+└──────────────────┼──────────────────────────────┘
+                   │
+                   ▼
+┌─────────────────────────────────────────────────┐
+│  Output                                          │
+│  - Success: "All requirements met"               │
+│  - Failure: List of missing items + instructions │
+└─────────────────────────────────────────────────┘
+```
+
+---
+
+## Alternatives Considered
+
+### Alternative 1: Dynamic Requirements from Devcontainer Config
+
+**Description:**
+
+Instead of hard-coding requirements, parse them from `.devcontainer/devcontainer.json` files in each language repo.
+
+**Pros:**
+
+- Single source of truth (devcontainer config)
+- Automatically stays in sync with container definitions
+- Leverages existing infrastructure
+
+**Cons:**
+
+- Devcontainer configs may not be complete or consistent across repos
+- Requires parsing different formats (some repos use Docker images, others use features)
+- Installation instructions not typically included in devcontainer configs
+- Adds complexity for a V1 tool
+
+**Why not chosen:**
+
+For V1, we want to prove the concept with minimal dependencies. V2 can explore dynamic discovery once we understand the usage patterns.
+
+---
+
+### Alternative 2: Check Requirements Only When Other Tools Fail
+
+**Description:**
+
+Don't provide a standalone verify-setup tool. Instead, when other tools fail (e.g., generate-sdk), detect if it's an environment issue and suggest specific fixes.
+
+**Pros:**
+
+- No separate tool needed
+- More contextual (only checks what's needed for the failing operation)
+- Simpler for developers (fewer commands to learn)
+
+**Cons:**
+
+- Reactive instead of proactive
+- Developers can't proactively check their setup before starting work
+- Each tool would need to implement error detection logic
+- Harder to onboard new contributors who want to verify setup upfront
+
+**Why not chosen:**
+
+A proactive verification tool provides better developer experience, especially for onboarding and when working across multiple languages.
+
+---
+
+### Alternative 3: Integration with Devcontainers CLI
+
+**Description:**
+
+Use the devcontainers CLI tool to build and validate container images locally, ensuring the environment matches CI.
+
+**Pros:**
+
+- Guarantees environment matches CI/CD
+- Comprehensive solution
+
+**Cons:**
+
+- Requires Docker and devcontainers CLI installed
+- Much heavier solution (downloading/building containers)
+- Not all developers want to work in containers
+- Slower feedback loop
+
+**Why not chosen:**
+
+Too heavy for a V1 environment verification tool. This could be a future option for developers who want container-based development.
+
+---
+
+## Open Questions
+
+- [ ] **Question 1**: How should we handle dynamic requirement discovery in V2?
+  - Context: Hard-coding requirements in V1 works but isn't maintainable long-term
```

Comment:
have you engaged with EngSys to see if they have some ideas on how this should happen? If not, that would be a good step for V2

Reply:
I added some ideas on this in https://github.com/Azure/azure-sdk-tools/pull/12457#issuecomment-3407751610 and based on the latest update to the notes it seems like the concept of a single source of truth has been discussed/brainstormed now?